yt_dlp/extractor/facebook.py

```diff
@@ -478,9 +480,35 @@ def _perform_login(self, username, password):
             return
 
     def _extract_from_url(self, url, video_id):
+        cookies = self._get_cookies(url)
+        # user passed logged-in cookies or attempted to login
+        login_data = cookies.get('c_user') and cookies.get('xs')
+        logged_in = False
+
         webpage = self._download_webpage(
             url.replace('://m.facebook.com/', '://www.facebook.com/'), video_id)
 
+        sjs_data = [self._parse_json(j, video_id, fatal=False) for j in re.findall(
+            r'data-sjs>({.*?ScheduledServerJS.*?})</script>', webpage)]
+
+        if login_data:
+            logged_in = get_first(sjs_data, (
+                'require', ..., ..., ..., '__bbox', 'define',
+                lambda _, v: 'CurrentUserInitialData' in v, ..., 'ACCOUNT_ID'), default='0') != '0'
```

Comment:
do we really need to check cookies if we are checking this value for login status?

Reply:
I feel we've lost a bit of context. Yes, we should as that indicates whether user `attempted` login in the first place. Since facebook would simply scrape the session from the `cookiejar` and appear as we've never tried to login.

While this also makes sure that passed cookies are also valid or not.